core/field_dropdown.js

```diff
@@ -36,698 +35,715 @@ const {Svg} = goog.require('Blockly.utils.Svg');
+   *    https://developers.google.com/blockly/guides/create-custom-blocks/fields/built-in-fields/dropdown#creation}
+   *    for a list of properties this parameter supports.
+   * @throws {TypeError} If `menuGenerator` options are incorrectly structured.
+   * @alias Blockly.FieldDropdown
    */
-  this.menuGenerator_ = menuGenerator;
+  constructor(menuGenerator, opt_validator, opt_config) {
+    super(Field.SENTINEL);
+
+    /**
+     * A reference to the currently selected menu item.
+     * @type {?MenuItem}
+     * @private
+     */
+    this.selectedMenuItem_ = null;
+
+    /**
+     * The dropdown menu.
+     * @type {?Menu}
+     * @protected
+     */
+    this.menu_ = null;
+
+    /**
+     * SVG image element if currently selected option is an image, or null.
+     * @type {?SVGImageElement}
+     * @private
+     */
+    this.imageElement_ = null;
+
+    /**
+     * Tspan based arrow element.
+     * @type {?SVGTSpanElement}
+     * @private
+     */
+    this.arrow_ = null;
+
+    /**
+     * SVG based arrow element.
+     * @type {?SVGElement}
+     * @private
+     */
+    this.svgArrow_ = null;
+
+    /**
+     * Serializable fields are saved by the serializer, non-serializable fields
+     * are not. Editable fields should also be serializable.
+     * @type {boolean}
+     */
+    this.SERIALIZABLE = true;
+
+    /**
+     * Mouse cursor style when over the hotspot that initiates the editor.
+     * @type {string}
+     */
+    this.CURSOR = 'default';
+
+
+    // If we pass sentinel, don't do *anything* with the menu generator.
+    if (menuGenerator == Field.SENTINEL) return;
+
+    if (Array.isArray(menuGenerator)) {
+      validateOptions(menuGenerator);
+    }
+
+    /**
+     * An array of options for a dropdown list,
+     * or a function which generates these options.
+     * @type {(!Array<!Array>|!function(this:FieldDropdown): !Array<!Array>)}
+     * @protected
+     */
+    this.menuGenerator_ =
+        /**
+@type {(!Array<!Array>|
+!function(this:FieldDropdown): !Array<!Array>)}
+*/ (menuGenerator);
+
+    /**
+     * A cache of the most recently generated options.
+     * @type {Array<!Array<string>>}
+     * @private
+     */
+    this.generatedOptions_ = null;
+
+    /**
+     * The prefix field label, of common words set after options are trimmed.
+     * @type {?string}
+     * @package
+     */
+    this.prefixField = null;
+
+    /**
+     * The suffix field label, of common words set after options are trimmed.
+     * @type {?string}
+     * @package
+     */
+    this.suffixField = null;
+
+    this.trimOptions_();
+
+    /**
+     * The currently selected option. The field is initialized with the
+     * first option selected.
+     * @type {!Object}
+     * @private
+     */
+    this.selectedOption_ = this.getOptions(false)[0];
+
+    /** @override */
+    this.value_ = this.selectedOption_[1];
+
+    if (opt_config) this.configure_(opt_config);
+    this.setValue(this.selectedOption_[1]);
```

Comment:
Answer to 1: That type annotation is super wrong, it should be `!Array<*>`, because `getOptions` is `!Array<!Array>

Answer to 2: I thought [override](https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#override) triggered annotation inheritance. Do I not need to do that?

Answers to 3 & 4:
This is just trying to follow the patterns of other Fields, which set a default value before calling `setValue` incase validation fails. The interesting thing about dropdown fields is that their initial value is guaranteed to be valid, because (1) the default option is the first option in the menu, and (2) a value is valid if it exists in the menu.

So this should just call `this.setValue(this.selectedOption_[1])` which I will do.

Reply:
> I thought [override](https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#override) triggered annotation inheritance. Do I not need to do that?

For methods that is recommended, but here I think you are just changing the value of the `.value_` property declared in the superclass, rather than overriding anything.